core-plugins/src/test/java/co/cask/hydrator/plugin/batch/action/SSHActionTest.java

```diff
@@ -0,0 +1,102 @@
+package co.cask.hydrator.plugin.batch.action;
+
+import co.cask.cdap.api.common.Bytes;
+import co.cask.hydrator.plugin.batch.ETLBatchTestBase;
+import co.cask.hydrator.plugin.common.EchoCommandFactory;
+import co.cask.hydrator.plugin.common.MockActionContext;
+import org.apache.commons.codec.binary.Base64;
+import org.apache.sshd.common.config.keys.KeyUtils;
+import org.apache.sshd.server.SshServer;
+import org.apache.sshd.server.auth.pubkey.PublickeyAuthenticator;
+import org.apache.sshd.server.keyprovider.SimpleGeneratorHostKeyProvider;
+import org.apache.sshd.server.session.ServerSession;
+import org.apache.sshd.server.shell.ProcessShellFactory;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.security.PublicKey;
+
+/**
+ * Test for {@link SSHAction}
+ */
+public class SSHActionTest extends ETLBatchTestBase {
+  // Test keypair config
+  private static final String PRIVATE_KEY =
+    "-----BEGIN RSA PRIVATE KEY-----\n" +
+    "Proc-Type: 4,ENCRYPTED\n" +
+    "DEK-Info: AES-128-CBC,9CEAA64BCD8E8EF8A953C0282D39FE1B\n" +
+    "\n" +
+    "9wFTqXKQkvtHnzthva5k+vv73DMjeRXckRIiSlJeY61kDbKLCGqQWc3DHqeQZikz\n" +
+    "vAAQFVUqMMX0MHt+mIJZa+7F1kYg6wo9jKr2BIKahwcdCJo5vDx/xBnz33VsGkWf\n" +
+    "gF2lo5X2qQIDrjer6SJLciop9jb91KIkbwE1uR+6/84EWigQQxn1a7rZEQxyZQK+\n" +
+    "wkeYfvWi5Ngav01v4Ui7KJA69+i8634WZrPvpNt1trIAPyW52EtnHA1MrOnKiWmh\n" +
+    "qCmniJ9LoVThnNJekQMCTXi01DZMuWxx6FcZaaJVu7jlxkaWea6l0Nt+gSnkdyxZ\n" +
+    "nb/DB/IC81gvQuUOxQ0gYlaZPkdCygRpFoWl2ssOuP1vBXqSdOeU61TFM45i4mJ9\n" +
+    "/YBB2nsDy6WWstSCLCNQatysVIQwX58Rt19sqHurOS2wOgB4BFhFux/Umd865jmO\n" +
+    "n74LeKM4V93mLpHjet2NF7MRnphOAf0en6cb1rAUFGSCVbVO9ogwxdvfHJcsHlTg\n" +
+    "cWeSMvTl+YvnTua93qth1l2weJA783BJrXBOd1no5UADDPP6DOxmC90VGHnIrI7/\n" +
+    "bmRxOxJeJbYHLKiTZ6AGcqhXN/G98uKVD20/hvBawf8=\n" +
+    "-----END RSA PRIVATE KEY-----";
```

Comment:
Another option is to programmatically generate a keypair for each test run but unsure if such a variable changing from run to run would be ideal.